### PR TITLE
Make license detection dynamic

### PR DESCRIPTION
**Pull request by jmsmkn**
This changes things around so that open source determination
is done at run time, rather than at import time. This allows
us to modify the list of licenses considered open source.

Licensee is upgraded to v9.15.1 so that json output is available.
This standardises the results and opens up more options such
as dynamically determining fine-grained restrictions such as
commercial use.

Also adds support for BSD licenses.

### CODE REVIEW

File: app/grandchallenge/github/models.py [hunk 1]
```diff
@@ -121,14 +120,29 @@ def user_error(self):
         else:
             return ""
 
+    @property
+    def license(self):
+        # Return the first license found in the results
+        return self.license_check_result.get("licenses", [{}])[0]
```
Comment (jmsmkn) [inline app/grandchallenge/github/models.py:126]: I don't know if this is super robust, we need to run it on a bunch of repos and find out.

Comment (jmsmkn) [reply app/grandchallenge/github/models.py:126]: Or, maybe check _all_ of the licenses found.

File: app/grandchallenge/github/migrations/0008_auto_20220201_1203.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+                    ("FAILURE", "Failure"),
+                    ("INVALID", "Invalid"),
+                    ("NOT_APPLICABLE", "Not Applicable"),
+                ],
+                default="NOT_APPLICABLE",
+                max_length=14,
+            ),
+        ),
+        migrations.RemoveField(
+            model_name="githubwebhookmessage", name="license_check_result",
```
Comment (jmsmkn) [inline app/grandchallenge/github/migrations/0008_auto_20220201_1203.py:32]: Note that I am removing the old field here and adding the JSON one back, will re-run the check on the existing messages.

File: app/grandchallenge/github/admin.py [hunk 3]
```diff
@@ -25,15 +25,12 @@ class GitHubWebhookMessageAdmin(admin.ModelAdmin):
...
-        "license_check_result",
         "user_error",
     )
-    list_filter = (
-        "clone_status",
-        "has_open_source_license",
-    )
-    search_fields = ("payload",)
+    list_filter = ("clone_status",)
+    search_fields = ("payload", "license_check")
```
Comment (MikeOverkamp-diag) [inline app/grandchallenge/github/admin.py:33]: Maybe I'm just overlooking something, but I cannot find the `license_check` field on the model?

File: app/grandchallenge/github/models.py (hunk 1 above)
Comment (MikeOverkamp-diag) [reply app/grandchallenge/github/models.py:126]: Maybe check all licenses and as long as there's an open source license, the check passes?

Comment (MikeOverkamp-diag) [reply app/grandchallenge/github/models.py:126]: Though I'd say multi-licensing is an edge case that might not be worth the added complexity.

Comment (jmsmkn) [reply app/grandchallenge/github/models.py:126]: Exactly, as this is dynamic we can always change it later. Lets see what data we get.

File: app/grandchallenge/github/admin.py (hunk 3 above)
Comment (jmsmkn) [reply app/grandchallenge/github/admin.py:33]: Good catch!

**Pull request merged**